lib/compress/zstd_compress.c

```diff
@@ -311,7 +311,7 @@ static ZSTD_CCtx_params ZSTD_makeCCtxParamsFromCParams(
     /* Adjust advanced params according to cParams */
     cctxParams.ldmParams.enableLdm = ZSTD_resolveEnableLdm(cctxParams.ldmParams.enableLdm, &cParams);
     if (cctxParams.ldmParams.enableLdm == ZSTD_ps_enable) {
-        ZSTD_ldm_adjustParameters(&cctxParams.ldmParams, &cParams);
+        ZSTD_ldm_adjustParameters(&cctxParams.ldmParams, &cctxParams);
```

Comment:
given that `cctxParams.ldmParams` is within `cctxParams`,
one has to wonder why it is initialised separately.
It looks to me that this is just a part of adjusting `cctxParams`, and should be bundled there,
this way there would be no reason to have separate actions in other places of the code.